src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,260 @@
+              simClimberGearbox,
+              ClimberConstants.KGearRatio,
+              SingleJointedArmSim.estimateMOI(
+                  ClimberConstants.kClimberArmLength.in(Meters), DrivetrainConstants.ROBOT_MASS_KG),
+              ClimberConstants.kClimberArmLength.in(Meters),
+              ClimberConstants.kClimberArmMinAngle.in(Radians),
+              ClimberConstants.kClimberArmMaxAngle.in(Radians),
+              true,
+              ClimberConstants.kClimberRetractedSetpoint.in(Radians));
+
+      climbMech = new Mechanism2d(2, 2);
+      climbMechRoot = climbMech.getRoot("Climb Mech root", 1, 1);
+      climbMechLigament =
+          climbMechRoot.append(
+              new MechanismLigament2d(
+                  "Climb Mech Ligament",
+                  1,
+                  Radians.of(climberSimNormal.getAngleRads()).in(Degrees)));
+      climbMechTargetLigament =
+          climbMechRoot.append(
+              new MechanismLigament2d(
+                  "Climb Mech Target Ligament",
+                  0.8,
+                  climberTargetAngle.in(Degrees),
+                  10,
+                  new Color8Bit(255, 255, 0)));
+      SmartDashboard.putData("Climb Mech", climbMech);
+    }
+  }
+
+  public Command zero() {
+    return runOnce(
+        () -> {
+          climberMotorFront.setPosition(
+              Degrees.of((climberFrontEncoder.get() + 90) * ClimberConstants.KGearRatio)
+                  .in(Rotations));
+          climberMotorBack.setPosition(
+              Degrees.of((climberBackEncoder.get() + 90) * ClimberConstants.KGearRatio)
+                  .in(Rotations));
+        });
+  }
+
+  public Command runRaw(Voltage voltage) {
+    return runOnce(
+        () -> {
+          climberMotorFront.setControl(new VoltageOut(voltage));
+          climberMotorBack.setControl(new VoltageOut(voltage.times(-1)));
+        });
+  }
+
+  private Command runClimber(Angle position, int slot) {
+    return runOnce(
+        () -> {
+          climberTargetAngle = position;
+          climberMotorFront.setControl(
+              new PositionVoltage(position.times(ClimberConstants.KGearRatio)).withSlot(slot));
+          climberMotorBack.setControl(
+              new PositionVoltage(position.times(ClimberConstants.KGearRatio)).withSlot(slot));
+          SmartDashboard.putNumber("Climber Position Setpoint", position.in(Degrees));
+        });
+  }
+
+  private BooleanSupplier isClimberAtPosition(Angle position) {
+    return () ->
+        position.isNear(
+            climberMotorFront.getPosition().getValue().div(ClimberConstants.KGearRatio),
+            ClimberConstants.kClimberSensorError.div(ClimberConstants.KGearRatio));
+  }
+
+  public Command climb() {
+    return Commands.sequence(
+        runClimber(ClimberConstants.kClimberAtCageSetpoint, 0),
+        Commands.waitUntil(isClimberAtPosition(ClimberConstants.kClimberAtCageSetpoint)),
+        runClimber(ClimberConstants.kClimberExtendedSetpoint, 1));
+  }
```

Comment:
I think we'll want to split this into two different Commands (i.e. two different buttons to control the sequence). That way its easier to retract/back off if the robot isn't in the right position during a match.